persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/CustomKeyGenerator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.cbioportal.persistence.util;
+
+import java.lang.reflect.Method;
+import org.cbioportal.persistence.CacheEnabledConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.util.StringUtils;
+
+public class CustomKeyGenerator implements KeyGenerator {
+
+    @Autowired
+    private CacheEnabledConfig cacheEnabledConfig;
+
+    private static final Logger LOG = LoggerFactory.getLogger(CustomKeyGenerator.class);
+
+    public Object generate(Object target, Method method, Object... params) {
+        if (!cacheEnabledConfig.isEnabled()) {
+            return "";
+        }
+        String key = target.getClass().getSimpleName() + "_"
+            + method.getName() + "_"
+            + StringUtils.arrayToDelimitedString(params, "_");
```

Comment:
We do have the cache statistics API endpoint which will report the set of current keys. I don't have statistics, but it must be true that there are some recurring queries which pass a list of tens of thousands of sample identifiers. However, often the response from the persistence layer will be fully modeled objects which also contain the sample ids plus additional information, so in general I would expect the size of the keys to be smaller than the size of the values. Still, it would be good to measure this in a public portal with the cache statistics endpoint turned on.

Reply:
we plan to have the endpoint turned on for the genie public portal